### ISSUE TITLE
[BUG] Fleet trying to uninstall monitoring and longhorn after upgrade from 2.6.6 to 2.6.7

### ISSUE THREAD

**Issue by erSitzt** [rancher/rancher#38672]
**Rancher Server Setup**
- Rancher version: 2.6.6 / 2.6.7
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version:
Local : v1.21.7+rke2r2
Downstream : v1.21.5+rke2r1 (imported RKE2)

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
Admin

**Describe the bug**
After upgrading Rancher from 2.6.6 to 2.6.7, which went without any problem on first glance, i noticed that somewhere around the time the cluster agent and fleet agent (not sure here) were updated my longhorn and monitoring apps were being uninstalled...
But not completely, they got stuck somewhere along the line with all other services using longhorn volumes going down as well...

Longhorn and Monitoring are Rancher-Charts and installed with minimal customization

Git Repo has this layout
<img width="209" alt="image" src="https://user-images.githubusercontent.com/1625376/185612552-728e49e9-4739-42b3-b772-770a7e8d9d2b.png">
longhorn-crd
```
defaultNamespace: rancher-logging
helm:
  releaseName: rancher-logging-crd
  repo: https://charts.rancher.io
  chart: rancher-logging-crd
  version: 100.0.1+up3.15.0
```

longhorn
```
defaultNamespace: longhorn-system
helm:
  repo: https://charts.rancher.io
  chart: longhorn
  releaseName: longhorn
  values:
    defaultSettings:
      createDefaultDiskLabeledNodes: true
      defaultDataPath: /longhorn-data/
      defaultReplicaCount: 2
      defaultDataLocality: best-effort
      replicaAutoBalance: best-effort
diff:
  comparePatches:
  - apiVersion: policy/v1beta1
    kind: PodSecurityPolicy
    operations:
    - {"op":"remove", "path":"/spec/hostIPC"}
    - {"op":"remove", "path":"/spec/hostNetwork"} 
  - apiVersion: v1
    kind: Service
    name: longhorn-frontend
    namespace: longhorn-system
    operations:
    - {"op":"remove", "path":"/spec/ports/0"}   

```

monitoring-crd
```
defaultNamespace: longhorn-system
helm:
  repo: https://charts.rancher.io
  chart: longhorn-crd
  releaseName: longhorn-crd
```

monitoring
```
defaultNamespace: cattle-monitoring-system
helm:
  releaseName: rancher-monitoring
  repo: https://charts.rancher.io
  chart: rancher-monitoring
  version: 100.1.0+up19.0.3

diff:
  comparePatches:
  - apiVersion: admissionregistration.k8s.io/v1beta1
    kind: MutatingWebhookConfiguration
    name: rancher-monitoring-admission
    operations:
    - {"op":"remove", "path":"/webhooks"}
  - apiVersion: admissionregistration.k8s.io/v1beta1
    kind: ValidatingWebhookConfiguration
    name: rancher-monitoring-admission
    jsonPointers:
    - "/webhooks"
  - apiVersion: admissionregistration.k8s.io/v1
    kind: MutatingWebhookConfiguration
    name: rancher-monitoring-admission
    operations:
    - {"op":"remove", "path":"/webhooks"}
  - apiVersion: admissionregistration.k8s.io/v1
    kind: ValidatingWebhookConfiguration
    name: rancher-monitoring-admission
    jsonPointers:
    - "/webhooks"    
  - apiVersion: policy/v1beta1
    kind: PodSecurityPolicy
    operations:
    - {"op":"remove", "path":"/spec/hostIPC"}
    - {"op":"remove", "path":"/spec/hostNetwork"}
    - {"op":"remove", "path":"/spec/hostPID"}
    - {"op":"remove", "path":"/spec/privileged"}
    - {"op":"remove", "path":"/spec/readOnlyRootFilesystem"}
  - apiVersion: apps/v1
    kind: Deployment
    name: rancher-monitoring-grafana
    namespace: cattle-monitoring-system
    operations:
    - {"op":"remove", "path":"/spec/template/spec/containers/0/env/0/value"}
  - apiVersion: apps/v1
    kind: Deployment
    operations:
    - {"op":"remove", "path":"/spec/template/spec/hostNetwork"}
    - {"op":"remove", "path":"/spec/template/spec/nodeSelector"}
    - {"op":"remove", "path":"/spec/template/spec/priorityClassName"}
    - {"op":"remove", "path":"/spec/template/spec/tolerations"}
  - apiVersion: v1
    kind: ServiceAccount
    operations:
    - {"op":"remove", "path":"/imagePullSecrets"}

```

These configs are more or less taken from these examples : https://github.com/ibrokethecloud/core-bundles

Not sure if this is a problem due to the update or a misconfig on my end or a problem with fleet...

This really sucks ! The cleanup/aftermath is no fun.. as Istio and Cert-Manager are depending on Monitoring and all the other stuff on Longhorn... :(

**Comment by xhejtman** [comment]
We saw this as well, but in our case, everything from fleet was just uninstalled :(

**Comment by xhejtman** [comment]
I also seems that a repo cannot contain more than one path. Only a single bundle is created even if repo contains multiple paths.

**Comment by xhejtman** [comment]
I think the actual problem is, that gitrepo containing more than 1 path (paths key) creates a bundle for the last one only and the other are uinstalled. 

@erSitzt can you confirm, you are using multiple paths within a single repo?

**Comment by erSitzt** [comment]
Yes, all those with crds

**Comment by xhejtman** [comment]
So, as a workaround, you need to switch to 1 path per gitrepo object. And btw, it seems that that `diff` part is ignored so everything will be in `modified` state.

**Comment by manno** [comment]
I'm trying to reproduce this issue, can you give me an example of the gitrepo CRD you are using (e.g.: `kubectl get gitrepo -A -o yaml`)?

**Comment by erSitzt** [comment]
@xhejtman it seems you are right...

This happens when i use a git repo with 
 - logging-crd
 - logging

<img width="929" alt="image" src="https://user-images.githubusercontent.com/1625376/186018460-924c77d4-d837-4a0f-b6c0-2cb59c7264c7.png">

Notice 9 Resources
.. and no CRDs
<img width="270" alt="image" src="https://user-images.githubusercontent.com/1625376/186018561-e746ed59-5f78-4117-b6af-c0697de6b34f.png">

And this after changing the order
<img width="933" alt="image" src="https://user-images.githubusercontent.com/1625376/186018674-6d7410fb-416c-45d5-b85b-39846fe01550.png">

<img width="623" alt="image" src="https://user-images.githubusercontent.com/1625376/186018724-ce755e17-f47c-429b-aecf-71375aa6a4de.png">


Funny sidenote... for a second it shows 14 Resources when updating...

**Comment by erSitzt** [comment]
This is the git-repo yaml
```
apiVersion: fleet.cattle.io/v1alpha1
kind: GitRepo
metadata:
  name: logging
  namespace: fleet-default
spec:
  branch: main
  clientSecretName: secret
  insecureSkipTLSVerify: false
  paths:
  - /rancher/logging-crd
  - /rancher/logging
  paused: false
  repo: https://srv-git.xxxx/kubernetes/environments/global-configs.git
  targets:
  - clusterGroup: logging
```

**Comment by xhejtman** [comment]
@erSitzt yes, it shows, because all the bundles do exist for a moment, but all except the last one are removed quite shortly.
I see something like this:
`2022-08-21T22:30:18.629850619+02:00 stderr F time="2022-08-21T20:30:18Z" level=debug msg="DesiredSet - Delete batch/v1, Kind=Job fleet-default/wes-auth-beb25 for sync-repo fleet-default/wes-auth"`

**Comment by erSitzt** [comment]
This is the third time i had major problems with fleet.... fun stuff :(
Far from production ready it seems.
And little to no feedback on the issues :(

**Comment by ronhorton** [comment]
**reproducible** 
1. created rancher HA with verison 2.6.6
2. imported rke2 cluster
3. click hamburger menu
4. select continuous delivery
5. Add git repo (attaching yaml at end of comment - showing error)
6. bundles install on downstream cluster
7. upgrade rancher to 2.6.7
8. after upgrade, checked downstream cluster
9. rancher logging was uninstalling
10. longhorn was uninstalling
11. monitoring was uninstalling
12. cluster also disappeared from fleet dashboard. 
<img width="1781" alt="2022-08-24_15-55-21" src="https://user-images.githubusercontent.com/89212856/186514048-6e8b60db-9187-4c04-bf8a-c7c4cb120c03.png">
<img width="1541" alt="2022-08-24_15-54-08" src="https://user-images.githubusercontent.com/89212856/186514233-036f55c7-55b4-4101-92ab-684a2dacdea9.png">
<img width="1755" alt="2022-08-24_15-53-47" src="https://user-images.githubusercontent.com/89212856/186514250-5c6b0522-1344-4413-8121-4b2b3c247c95.png">
<img width="1562" alt="2022-08-24_15-54-44" src="https://user-images.githubusercontent.com/89212856/186514276-1e3a5d9f-957a-4b1a-91e1-33a48d502eb2.png">

fleet-controller log:
```time="2022-08-24T19:52:58Z" level=info msg="No access to list CRDs, assuming CRDs are pre-created."
I0824 19:52:58.206509       1 leaderelection.go:248] attempting to acquire leader lease cattle-fleet-system/fleet-controller-lock...
I0824 19:53:01.308370       1 leaderelection.go:258] successfully acquired lease cattle-fleet-system/fleet-controller-lock
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=BundleNamespaceMapping controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=Cluster controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting /v1, Kind=ConfigMap controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting rbac.authorization.k8s.io/v1, Kind=Role controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting rbac.authorization.k8s.io/v1, Kind=RoleBinding controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=ClusterRegistration controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=ImageScan controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=Bundle controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting gitjob.cattle.io/v1, Kind=GitJob controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=BundleDeployment controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting /v1, Kind=Secret controller"
time="2022-08-24T19:53:01Z" level=info msg="All controllers have been started"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=GitRepoRestriction controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=ClusterRegistrationToken controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=ClusterGroup controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting rbac.authorization.k8s.io/v1, Kind=ClusterRole controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=Content controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting /v1, Kind=ServiceAccount controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting fleet.cattle.io/v1alpha1, Kind=GitRepo controller"
time="2022-08-24T19:53:01Z" level=info msg="Starting /v1, Kind=Namespace controller"
time="2022-08-24T19:53:11Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:21304c5d3a192ccab01648923d44e3151ef8a753]) 1.0015581s"
time="2022-08-24T19:53:12Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:e5e9472161bbb7845a06f5568baf797e93531561]) 1.000420026s"
time="2022-08-24T19:53:20Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:e5e9472161bbb7845a06f5568baf797e93531561]) 1.003909505s"
time="2022-08-24T19:53:26Z" level=error msg="error syncing 'fleet-default/ibroke-multi-apps-longhorn-crd': handler bundle-display: Operation cannot be fulfilled on bundles.fleet.cattle.io \"ibroke-multi-apps-longhorn-crd\": StorageError: invalid object, Code: 4, Key: /registry/fleet.cattle.io/bundles/fleet-default/ibroke-multi-apps-longhorn-crd, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 39ae9d75-0458-4e40-b8a2-a0eba339b759, UID in object meta: , requeuing"
time="2022-08-24T19:53:26Z" level=error msg="error syncing 'fleet-default/ibroke-multi-apps-longhorn-crd': handler bundle-display: Operation cannot be fulfilled on bundles.fleet.cattle.io \"ibroke-multi-apps-longhorn-crd\": StorageError: invalid object, Code: 4, Key: /registry/fleet.cattle.io/bundles/fleet-default/ibroke-multi-apps-longhorn-crd, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 39ae9d75-0458-4e40-b8a2-a0eba339b759, UID in object meta: , requeuing"
time="2022-08-24T19:53:29Z" level=error msg="error syncing 'cluster-fleet-default-ron-rke2-2-e94d50df4853/ibroke-multi-apps-monitoring': handler bundledeployment-display: Operation cannot be fulfilled on bundledeployments.fleet.cattle.io \"ibroke-multi-apps-monitoring\": StorageError: invalid object, Code: 4, Key: /registry/fleet.cattle.io/bundledeployments/cluster-fleet-default-ron-rke2-2-e94d50df4853/ibroke-multi-apps-monitoring, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: d76189d6-f4c6-4cd1-bbdf-bee24f43253c, UID in object meta: , requeuing"
time="2022-08-24T19:53:30Z" level=error msg="error syncing 'fleet-default/ibroke-multi-apps-monitoring': handler bundle: Operation cannot be fulfilled on bundles.fleet.cattle.io \"ibroke-multi-apps-monitoring\": StorageError: invalid object, Code: 4, Key: /registry/fleet.cattle.io/bundles/fleet-default/ibroke-multi-apps-monitoring, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 29456d5a-e701-4f7e-a8d2-1022e079c5d1, UID in object meta: , handler bundle-display: Operation cannot be fulfilled on bundles.fleet.cattle.io \"ibroke-multi-apps-monitoring\": StorageError: invalid object, Code: 4, Key: /registry/fleet.cattle.io/bundles/fleet-default/ibroke-multi-apps-monitoring, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 29456d5a-e701-4f7e-a8d2-1022e079c5d1, UID in object meta: , requeuing"
time="2022-08-24T19:55:11Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:1ebcdc82573204eaeb320fdf6f3cec0102a883bf]) 1.002474651s"
time="2022-08-24T19:55:23Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:e5e9472161bbb7845a06f5568baf797e93531561]) 1.007406668s"
time="2022-08-24T19:55:29Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:1ebcdc82573204eaeb320fdf6f3cec0102a883bf]) 1.002726113s"
time="2022-08-24T19:56:30Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:e5e9472161bbb7845a06f5568baf797e93531561]) 1.002378702s"
time="2022-08-24T19:56:33Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:1ebcdc82573204eaeb320fdf6f3cec0102a883bf]) 1.004446522s"
time="2022-08-24T19:56:36Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:e5e9472161bbb7845a06f5568baf797e93531561]) 1.002741208s"
time="2022-08-24T19:56:39Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:e5e9472161bbb7845a06f5568baf797e93531561]) 1.0006832s"
time="2022-08-24T19:56:39Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:1ebcdc82573204eaeb320fdf6f3cec0102a883bf]) 1.002770839s"
time="2022-08-24T19:56:47Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:1ebcdc82573204eaeb320fdf6f3cec0102a883bf]) 1.000543647s"
time="2022-08-24T19:57:35Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:e5e9472161bbb7845a06f5568baf797e93531561]) 1.003277543s"
time="2022-08-24T19:57:58Z" level=info msg="Marking orphaned content[s-f5492d8617e7eca62e4c3a92f03b7763fee5a5770ef6dc9be2c29f6c22c46] for deletion"
time="2022-08-24T19:57:58Z" level=info msg="Marking orphaned content[s-6f79a666859174cd895f26c79e3157d190e5716a03e3c21f4ee444e6732a3] for deletion"
time="2022-08-24T19:57:58Z" level=info msg="Marking orphaned content[s-aeed0b88cf4c050c0e5c0809ae35056def5afe14a1daa0f0a7141d281190d] for deletion"
time="2022-08-24T19:57:58Z" level=info msg="Marking orphaned content[s-b2482f1772f87480a52c63d0b01e65a13696b5632692160fe53d4ac04ff80] for deletion"
time="2022-08-24T19:57:58Z" level=info msg="Marking orphaned content[s-b54ef07a7bc96a575544835cf66b7a286df43d2f27ff23f0d465efc4e3ea0] for deletion"
time="2022-08-24T19:57:58Z" level=info msg="Marking orphaned content[s-d33f3ac2b3ff72448a9ce0c06498fb372b5f6fc05171e45fb7ff3be94f763] for deletion"
time="2022-08-24T19:57:58Z" level=info msg="Marking orphaned content[s-fab06407231317de94b69477c781a86791ab601ce03c950e2a8a9fd3d51d1] for deletion"
time="2022-08-24T19:58:09Z" level=info msg="rate limited cluster-group-token(map[objectset.rio.cattle.io/hash:e5e9472161bbb7845a06f5568baf797e93531561]) 1.000309733s"```

**git repo yaml**


apiVersion: fleet.cattle.io/v1alpha1
kind: GitRepo
metadata:
  creationTimestamp: "2022-08-24T19:33:39Z"
  generation: 2
  managedFields:
  - apiVersion: fleet.cattle.io/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
      f:spec:
        .: {}
        f:branch: {}
        f:forceSyncGeneration: {}
        f:insecureSkipTLSVerify: {}
        f:paths: {}
        f:repo: {}
        f:targets: {}
    manager: rancher
    operation: Update
    time: "2022-08-24T19:37:02Z"
  - apiVersion: fleet.cattle.io/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        .: {}
        f:commit: {}
        f:conditions: {}
        f:desiredReadyClusters: {}
        f:display:
          .: {}
          f:error: {}
          f:message: {}
          f:readyBundleDeployments: {}
          f:state: {}
        f:gitJobStatus: {}
        f:lastSyncedImageScanTime: {}
        f:observedGeneration: {}
        f:readyClusters: {}
        f:resourceCounts:
          .: {}
          f:desiredReady: {}
          f:missing: {}
          f:modified: {}
          f:notReady: {}
          f:orphaned: {}
          f:ready: {}
          f:unknown: {}
          f:waitApplied: {}
        f:resources: {}
        f:summary:
          .: {}
          f:desiredReady: {}
          f:errApplied: {}
          f:nonReadyResources: {}
          f:ready: {}
    manager: fleetcontroller
    operation: Update
    subresource: status
    time: "2022-08-24T19:54:26Z"
  name: ibroke-multi-apps
  namespace: fleet-default
  resourceVersion: "71542"
  uid: 8a0d82cc-1ca1-4735-aec9-6e412744a2d5
spec:
  branch: master
  forceSyncGeneration: 2
  insecureSkipTLSVerify: true
  paths:
  - /logging-crd/
  - /logging/
  - /monitoring-crd/
  - /monitoring/
  - /longhorn-crd/
  - /longhorn/
  repo: https://github.com/ibrokethecloud/core-bundles/
  targets:
  - clusterName: ron-rke2-2
status:
  commit: 0d134b49b033fa6dc0600dc19b490be1265c7693
  conditions:
  - lastUpdateTime: "2022-08-24T19:54:26Z"
    message: 'ErrApplied(1) [Bundle ibroke-multi-apps-longhorn: execution error at
      (longhorn/templates/validate-install-crd.yaml:27:7): Required CRDs are missing.
      Please install the corresponding CRD chart before installing this chart.]'
    status: "False"
    type: Ready
  - lastUpdateTime: "2022-08-24T20:15:45Z"
    status: "True"
    type: Accepted
  - lastUpdateTime: "2022-08-24T19:33:39Z"
    status: "True"
    type: ImageSynced
  - lastUpdateTime: "2022-08-24T19:52:54Z"
    status: "False"
    type: Reconciling
  - lastUpdateTime: "2022-08-24T19:33:39Z"
    status: "False"
    type: Stalled
  - lastUpdateTime: "2022-08-24T20:15:45Z"
    status: "True"
    type: Synced
  desiredReadyClusters: 1
  display:
    error: true
    message: 'execution error at (longhorn/templates/validate-install-crd.yaml:27:7):
      Required CRDs are missing. Please install the corresponding CRD chart before
      installing this chart.'
    readyBundleDeployments: 0/1
    state: ErrApplied
  gitJobStatus: Current
  lastSyncedImageScanTime: null
  observedGeneration: 2
  readyClusters: 0
  resourceCounts:
    desiredReady: 22
    missing: 0
    modified: 0
    notReady: 22
    orphaned: 0
    ready: 0
    unknown: 0
    waitApplied: 0
  resources:
  - apiVersion: apps/v1
    id: longhorn-system/longhorn-manager
    kind: DaemonSet
    name: longhorn-manager
    namespace: longhorn-system
    state: ErrApplied
    type: apps.daemonset
  - apiVersion: apps/v1
    id: longhorn-system/longhorn-admission-webhook
    kind: Deployment
    name: longhorn-admission-webhook
    namespace: longhorn-system
    state: ErrApplied
    type: apps.deployment
  - apiVersion: apps/v1
    id: longhorn-system/longhorn-conversion-webhook
    kind: Deployment
    name: longhorn-conversion-webhook
    namespace: longhorn-system
    state: ErrApplied
    type: apps.deployment
  - apiVersion: apps/v1
    id: longhorn-system/longhorn-driver-deployer
    kind: Deployment
    name: longhorn-driver-deployer
    namespace: longhorn-system
    state: ErrApplied
    type: apps.deployment
  - apiVersion: apps/v1
    id: longhorn-system/longhorn-ui
    kind: Deployment
    name: longhorn-ui
    namespace: longhorn-system
    state: ErrApplied
    type: apps.deployment
  - apiVersion: v1
    id: longhorn-system/longhorn-default-setting
    kind: ConfigMap
    name: longhorn-default-setting
    namespace: longhorn-system
    state: ErrApplied
    type: configmap
  - apiVersion: v1
    id: longhorn-system/longhorn-storageclass
    kind: ConfigMap
    name: longhorn-storageclass
    namespace: longhorn-system
    state: ErrApplied
    type: configmap
  - apiVersion: policy/v1beta1
    id: longhorn-psp
    kind: PodSecurityPolicy
    name: longhorn-psp
    state: ErrApplied
    type: policy.podsecuritypolicy
  - apiVersion: rbac.authorization.k8s.io/v1
    id: longhorn-admin
    kind: ClusterRole
    name: longhorn-admin
    state: ErrApplied
    type: rbac.authorization.k8s.io.clusterrole
  - apiVersion: rbac.authorization.k8s.io/v1
    id: longhorn-edit
    kind: ClusterRole
    name: longhorn-edit
    state: ErrApplied
    type: rbac.authorization.k8s.io.clusterrole
  - apiVersion: rbac.authorization.k8s.io/v1
    id: longhorn-role
    kind: ClusterRole
    name: longhorn-role
    state: ErrApplied
    type: rbac.authorization.k8s.io.clusterrole
  - apiVersion: rbac.authorization.k8s.io/v1
    id: longhorn-view
    kind: ClusterRole
    name: longhorn-view
    state: ErrApplied
    type: rbac.authorization.k8s.io.clusterrole
  - apiVersion: rbac.authorization.k8s.io/v1
    id: longhorn-bind
    kind: ClusterRoleBinding
    name: longhorn-bind
    state: ErrApplied
    type: rbac.authorization.k8s.io.clusterrolebinding
  - apiVersion: rbac.authorization.k8s.io/v1
    id: longhorn-system/longhorn-psp-role
    kind: Role
    name: longhorn-psp-role
    namespace: longhorn-system
    state: ErrApplied
    type: rbac.authorization.k8s.io.role
  - apiVersion: rbac.authorization.k8s.io/v1
    id: longhorn-system/longhorn-psp-binding
    kind: RoleBinding
    name: longhorn-psp-binding
    namespace: longhorn-system
    state: ErrApplied
    type: rbac.authorization.k8s.io.rolebinding
  - apiVersion: v1
    id: longhorn-system/longhorn-admission-webhook
    kind: Service
    name: longhorn-admission-webhook
    namespace: longhorn-system
    state: ErrApplied
    type: service
  - apiVersion: v1
    id: longhorn-system/longhorn-backend
    kind: Service
    name: longhorn-backend
    namespace: longhorn-system
    state: ErrApplied
    type: service
  - apiVersion: v1
    id: longhorn-system/longhorn-conversion-webhook
    kind: Service
    name: longhorn-conversion-webhook
    namespace: longhorn-system
    state: ErrApplied
    type: service
  - apiVersion: v1
    id: longhorn-system/longhorn-engine-manager
    kind: Service
    name: longhorn-engine-manager
    namespace: longhorn-system
    state: ErrApplied
    type: service
  - apiVersion: v1
    id: longhorn-system/longhorn-frontend
    kind: Service
    name: longhorn-frontend
    namespace: longhorn-system
    state: ErrApplied
    type: service
  - apiVersion: v1
    id: longhorn-system/longhorn-replica-manager
    kind: Service
    name: longhorn-replica-manager
    namespace: longhorn-system
    state: ErrApplied
    type: service
  - apiVersion: v1
    id: longhorn-system/longhorn-service-account
    kind: ServiceAccount
    name: longhorn-service-account
    namespace: longhorn-system
    state: ErrApplied
    type: serviceaccount
  summary:
    desiredReady: 1
    errApplied: 1
    nonReadyResources:
    - bundleState: ErrApplied
      message: 'execution error at (longhorn/templates/validate-install-crd.yaml:27:7):
        Required CRDs are missing. Please install the corresponding CRD chart before
        installing this chart.'
      name: ibroke-multi-apps-longhorn
    ready: 0

**Comment by ronhorton** [comment]
**PASS Verified Fixed `2.6.8-rc1`**

rancher upgrade from `2.6.6 - 2.6.8-rc1`
1. rancher updated to 2.6.8-rc1
2. fleet pods restarted in local cluster
3. downstream cluster apps are present and running
4. fleet pods restarted on downstream cluster
5. new fleet image `v0.3.11-rc1` pulled, container created

local cluster:
<img width="961" alt="2022-08-27_14-05-23" src="https://user-images.githubusercontent.com/89212856/187043178-82ca816e-c960-437f-ab45-b2976c13387e.png">

downstream cluster:
<img width="1349" alt="2022-08-27_14-02-59" src="https://user-images.githubusercontent.com/89212856/187043175-2b01679b-c4f4-44f9-a4ca-3c6d7b906604.png">

**Comment by xhejtman** [comment]
Any chance to release 2.6.7-patch1 including this fix?

**Comment by aiyengar2** [comment]
The fix has been shipped in 2.6.8
